### PR TITLE
KAA-671: Fix. Implemented supervision strategy for main actors

### PR DESCRIPTION
**Pull request by ashvayka**

### CODE REVIEW

**Comment by ashvayka** [comment]
@Shpigor, @batytskyy  please review

File: pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@ Copyright 2014-2015 CyberVision, Inc.
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <main.dir>${basedir}</main.dir>
-        <akka.version>2.3.1</akka.version>
+        <akka.version>2.3.14</akka.version>
```
Comment (shpigor) [inline pom.xml:33]: :thumbsup:

**Comment by shpigor** [comment]
Reviewed

File: server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/ApplicationActor.java [hunk 2]
```diff
@@ -106,6 +107,11 @@ private ApplicationActor(AkkaContext context, String applicationToken) {
         this.userVerifierActor = getOrCreateUserVerifierActor();
     }
 
+    @Override
+    public SupervisorStrategy supervisorStrategy() {
+      return SupervisionStrategyFactory.createApplicationActorStrategy(context);
```
Comment (batytskyy) [inline server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/ApplicationActor.java]: Please fix the indentation to be 4 spaces.

File: server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/OperationsServerActor.java [hunk 3]
```diff
@@ -70,6 +72,11 @@ private OperationsServerActor(AkkaContext context) {
         this.statusRequestStatesMap = new HashMap<UUID, StatusRequestState>();
     }
 
+    @Override
+    public SupervisorStrategy supervisorStrategy() {
+      return SupervisionStrategyFactory.createOpsActorStrategy(context);
```
Comment (batytskyy) [inline server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/OperationsServerActor.java]: Please fix the indentation to be 4 spaces.

File: server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/TenantActor.java [hunk 4]
```diff
@@ -86,6 +86,11 @@ private TenantActor(AkkaContext context, String tenantId) {
         this.statusRequestStatesMap = new HashMap<UUID, StatusRequestState>();
     }
 
+    @Override
+    public SupervisorStrategy supervisorStrategy() {
+      return SupervisionStrategyFactory.createTenantActorStrategy(context);
```
Comment (batytskyy) [inline server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/TenantActor.java]: Please fix the indentation to be 4 spaces.

File: server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/cache/concurrent/CacheTemporaryMemorizer.java [hunk 5]
```diff
@@ -46,6 +47,9 @@
      * @return the v
      */
     public V compute(final K key, final Computable<K, V> worker){
+        if(key == null){
```
Comment (batytskyy) [inline server/operations/src/main/java/org/kaaproject/kaa/server/operations/service/cache/concurrent/CacheTemporaryMemorizer.java]: Please use spaces around if brackets.

**Comment by batytskyy** [comment]
Minor code-indentation omissions, otherwise LGTM.

**Pull request merged**